pkg/spanconfig/spanconfigkvsubscriber/kvsubscriber.go

```diff
@@ -206,9 +243,72 @@ func New(
 //	the exported StoreReader will be up-to-date and continue to be
 //	incrementally maintained.
 func (s *KVSubscriber) Start(ctx context.Context, stopper *stop.Stopper) error {
+	if err := stopper.RunAsyncTask(ctx, "kvsubscriber-metrics",
+		func(ctx context.Context) {
+			settingChangeCh := make(chan struct{}, 1)
+			metricsPollerInterval.SetOnChange(
```

Comment:
this set on change business... I is it really needed? There seems to be some dislike of that facility.  You could just get the value, and if it's 0 (disabled), recheck once every minute or 5.... It really doesn't matter.  And if you check infrequently enough, then ... perhaps you could just reset based on the setting value on each iteration.

Reply:
I'm not sure who dislikes it, I like it. It's helpful to have reactive settings -- if I've set it to 10m and then want to undo it, it's nice to just change it to whatever and have it take effect then. It feels easier than introducing two timers.